mlflow/tracing/utils/otlp.py

```diff
@@ -0,0 +1,61 @@
+import os
+from typing import Optional
+
+from opentelemetry.sdk.trace.export import SpanExporter
+
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import RESOURCE_DOES_NOT_EXIST
+
+
+def should_use_otlp_exporter() -> bool:
+    return _get_otlp_endpoint() is not None
+
+
+def get_otlp_exporter() -> SpanExporter:
+    """
+    Get the OTLP exporter based on the configured protocol.
+    """
+    endpoint, protocol = _get_otlp_endpoint(), _get_otlp_protocol()
+    if protocol == "grpc":
+        try:
+            from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
+        except ImportError:
+            raise MlflowException(
+                "gRPC OTLP exporter is not available. Please install the required dependency by "
+                "running `pip install opentelemetry-exporter-otlp-proto-grpc`.",
+                error_code=RESOURCE_DOES_NOT_EXIST,
+            )
+
+        return OTLPSpanExporter(endpoint=endpoint)
+    elif protocol == "http/protobuf":
+        try:
+            from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+        except ImportError as e:
+            raise MlflowException(
+                "HTTP OTLP exporter is not available. Please install the required dependency by "
+                "running `pip install opentelemetry-exporter-otlp-proto-http`.",
+                error_code=RESOURCE_DOES_NOT_EXIST,
+            ) from e
+
+        return OTLPSpanExporter(endpoint=endpoint)
+    else:
+        raise MlflowException.invalid_parameter_value(
+            f"Unsupported OTLP protocol '{protocol}' is configured. Please set "
+            "OTEL_EXPORTER_OTLP_TRACES_PROTOCOL env variable to either 'grpc' or 'http/protobuf'."
```

Comment:
Is both OTEL_EXPORTER_OTLP_TRACES_PROTOCOL and OTEL_EXPORTER_OPTL_PROTOCOL acceptable?

Reply:
Ah true, indeed we don't have to mention env var name here (cuz this only raises when user set either of them). Will rephrase it !